segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/HDFSExceptionHelpers.java

```diff
@@ -30,26 +30,23 @@
      *
      * @param segmentName Name of the stream segment on which the exception occurs.
      * @param e           The exception to be translated.
-     * @return Nothing. This method always throws.
+     * @return  The exception to be thrown.
      */
-    static <T> T throwException(String segmentName, Throwable e) throws StreamSegmentException {
+    static <T> StreamSegmentException throwException(String segmentName, Throwable e) {
```

Comment:
1. This method has inconsistent behavior. If it's one of the expected exceptions, it returns a new exception. If not, it quietly re-throws the original exception. It should be either one or the other.
2. If you decide to actually translate the exceptions instead of re-throwing them, then:
    - You should rename this to `convertException`. It doesn't throw anything anymore.
    - You should update the Javadoc